### DISCUSSION TITLE
Request Help : Running OpenWebui And Ollama On Proxmox > LXC > Docker

### DISCUSSION THREAD

**Discussion by GhassanYusuf** [open-webui/open-webui#6266] [General]
Dear Friends!  

I'm thrilled to share with you my latest project - a home lab server that I've lovingly built from scratch! 

Meet the Beast:  

    🔥 CPU: Intel Xeon E5 2680 V4 
        12 Cores / 24 Threads for seamless multitasking
        Turbo Boost up to 3.3 GHz for lightning-fast processing
         
    🚀 GPU: NVIDIA GeForce RTX 3060 12GB 
        Ray tracing and AI-enhanced graphics for a new level of realism
        12 GB GDDR6 memory for fast and smooth rendering
        PCIe 4.0 x16 interface for optimal performance
         
    🤩 RAM: 32GB DDR4 @ 2400MHz 
        Plenty of memory for running multiple applications simultaneously
        Fast data transfer rates for improved performance
         
    🔒 Motherboard: Machinist X99 K9 
        High-quality audio capabilities (5.1 channel ALC897 sound card)
        Multiple USB ports and SATA ports for connecting peripherals
        Support for PCIe expansion cards

The Goal:  Run multiple VMs, including a Linux container with Ollama on an LXC, where I want to install Docker, Open Web UI, and Nvidia support! 

And Here's What I Did:  

1. GPU Pass-Through : Set up LXC GPU pass-through for seamless performance.
2. Nvidia Drivers & Cuda Cores : Installed the latest Nvidia drivers and CUDA cores for optimal graphics performance.
3. Docker & Docker Container Toolkit : Installed Docker and its companion tool kit to enable containerized applications.
4. Ollama Installation : Installed Ollama, the AI-powered model training software.
5. Open Web UI : Installed Open Web UI with Nvidia support.

The Issue:  When I tried to register on Open Web UI and pull models from the settings, it gave me an error - "**Open WebUI: Server Connection Error**" followed by a "**Download Canceled**" message. 

Can You Help?  

If any of you have encountered similar issues or have solutions for this problem, please share your expertise! I'd be forever grateful for your help in resolving this mystery.

**Comment by Aducklab** [comment]
I'm just now starting to try to work with lxc containers, but the error you are getting is the same one I was getting when I was out of space on the drive I had llama and open web ui installed on in an Ubuntu server. I had to increase the drive storage to be able to download the models

**Reply by PainPlay** [reply]
I'm running OpenWEBUI on an LXC for already 9 months, with passing through my RTX A4000. Works like a charm. When I started, I had the same issue like you do, and it was all related to storage capacity. I'm gave it around 250GiB of storage, and that is more then enough, as you will soon learn that you'll stick to certain models and fine tune them to your liking.

So first increase your storage and let us know if that helped.


**As for what you otherwise mentioned:**
> running multiple VMs: 

Just keep in mind that you can only do so much with your vcores (24 of them), and that 32GB of memory is maybe a bit on the low end. VM's tend to require more RAM (depending on the VM/ OS you're using), but just keep this in the back of your head. Also, pass through is only capable to 1 VM, not multiple, or to multiple LXC but NO VM then. so if you where planning on using your GPU for multiple VM's, you're out of luck. There are some really sketchy work arounds our there on proxmox, but 99% of them fail or break your system.

EDIT: I just seen you're also talking about AI-graphics, so I assume you'll go for something like Stable Diffusion or so.
If that is the case, you can also run that in a separated LXC (again, one GPU can be passed though to multiple LXC's), what give you the advantage of well, having separated LXC's for different use cases. Just keep in mind that setting up access to the output files can be a bit tricky, rights related, but if you're on a Windowd machine, you can set up SMB and then map your LXC (or preferred folder) as a network drive. The only thing is then the rwx rights that for whatever reason don't carry over to now generated folders, but that's just the case of running one command line and be done with it.

**Comment by havardthom** [comment]
Open WebUI LXC script with optional Ollama install has been added to Proxmox Helper-Scripts: https://tteck.github.io/Proxmox/#open-webui-lxc

Alternatively there is also a separate Ollama LXC script which builds Ollama from source and supports Intel iGPU passthrough (very long install time): `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/ollama.sh)"`

**Reply by milanow** [reply]
I just play with LXC for 2 weeks, wonder have you started Pipelines on another LXC? It is possible right I guess? I notice that it requires 4G RAM, what about Pipelines do you have any suggestions? Thanks

**Reply by Soul-OMON** [reply]
> Alternatively there is also a separate Ollama LXC script which builds Ollama from source and supports Intel iGPU passthrough (very long install time): `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/ollama.sh)"`

Loading...
 ✗ This version of Proxmox Virtual Environment is not supported
Requires Proxmox Virtual Environment Version 8.1 or later.
Exiting...
